File: classical_algorithms/classic_algorithm_cpp/insertion_sort.cpp
#include<iostream>
using namespace std;
int n;

void insertion_sort(int a[])
{
    int hole,value;
    for(int i=1;i<n;i++)
    {
        hole=i;
        value=a[i];
        while(a[hole-1]>value && hole>0)
        {
            a[hole]=a[hole-1];
            hole=hole-1;
        }
        a[hole]=value;
    }
   cout<<"Sorted elements list: ";
   for(int i=0;i<n;i++)
      cout<<a[i]<<" ";
}
int main()
{
    cout<<"enter the size of the array = ";
  cin>>n;
  int a[n] = {0};
  
  cout<<"enter all the array element to sort them = ";
   for(int i=0;i<n;i++)
   {
       cin>>a[i];
   }
   insertion_sort(a);

    return 0;
}

File: classical_algorithms/python/Insertion Sort.py
def insertion_sort(array):
    
    for i in range(1,len(array)):
        j = i-1
        key = array[i]
        
        #move all the elements of array[0,i-1] that are greater than
        #the key, one position ahead of their current one
        
        while j>=0 and array[j]>key:
            array[j+1] = array[j]
            j = j-1
        
        array[j+1] = key
    
    print(array)
    

m = int(input('Enter the number of elements in the array '))

mylist = []

for i in range(0,m):
    num = int(input('Enter element: '))
    mylist.append(num)

insertion_sort(mylist)